### PR TITLE
Add namespace support.

### PR DESCRIPTION
**Pull request by petermattis**
Added DB.{CreateNamespace,ListNamespaces}. Table names are now required
to have the form "<namespace>.<table>".

### CODE REVIEW

File: client/table.go [hunk 1]
```diff
@@ -205,15 +205,80 @@ func (m *model) encodeColumnKey(primaryKey []byte, colID uint32) []byte {
 	return roachencoding.EncodeUvarint(key, uint64(colID))
 }
 
+// CreateNamespace creates a new namespace.
+//
+// TODO(pmattis): Is "namespace" the correct terminology? PostgreSQL and MySQL
+// use "database". PostgreSQL also has the notion of a schema. A
+// fully-qualified name in PostgreSQL looks like
+// "<database>.<schema>.<table>". See
+// http://www.postgresql.org/docs/9.4/static/ddl-schemas.html.
```
Comment (vivekmenezes) [inline client/table.go]: oracle does the same. I think "database" is meant to be the place where you configure resource requirements and such. Schema is the place to describe your problem domain as tables and constraints.

File: client/table.go [hunk 2]
```diff
@@ -205,15 +205,80 @@ func (m *model) encodeColumnKey(primaryKey []byte, colID uint32) []byte {
...
+	for i, row := range rows {
+		names[i] = string(bytes.TrimPrefix(row.Key, prefix))
+	}
+	return names, nil
+}
+
+func (db *DB) lookupTable(path string) (nsID uint32, name string, err error) {
+	parts := strings.Split(strings.ToLower(path), ".")
+	if len(parts) != 2 {
+		return 0, "", fmt.Errorf("malformed path: %s", path)
```
Comment (vivekmenezes) [inline client/table.go]: suggest how to fix it in the error

**Comment by vivekmenezes** [comment]
LGTM

File: client/table.go (hunk 2 above)
Comment (petermattis) [inline client/table.go]: Done.

**Pull request merged**